src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java

```diff
@@ -445,13 +456,50 @@
         @Qualifier("hearingBundleReadyNotificationGenerator") List<NotificationGenerator> notificationGenerators) {
 
         return new NotificationHandler(
-            (callbackStage, callback) ->
-                callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
-                    && callback.getEvent() == Event.GENERATE_HEARING_BUNDLE,
+            (callbackStage, callback) -> {
+
+                final String stitchStatus = getStitchStatus(callback);
+
+                return
+                    callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+                        && callback.getEvent() == Event.ASYNC_STITCHING_COMPLETE
+                        && stitchStatus.equalsIgnoreCase("DONE");
+            },
+            notificationGenerators
+        );
+    }
+
+    @Bean
+    public PreSubmitCallbackHandler<AsylumCase> hearingBundleFailedNotificationHandler(
+        @Qualifier("hearingBundleFailedNotificationGenerator") List<NotificationGenerator> notificationGenerators) {
+
+        return new NotificationHandler(
+            (callbackStage, callback) -> {
+
+                final String stitchStatus = getStitchStatus(callback);
+
+                return
+                    callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+                        && callback.getEvent() == Event.ASYNC_STITCHING_COMPLETE
+                        && stitchStatus.equalsIgnoreCase("FAILED");
+            },
             notificationGenerators
         );
     }
 
+    private String getStitchStatus(Callback<AsylumCase> callback) {
+        AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+        Optional<List<IdValue<Bundle>>> maybeCaseBundles = asylumCase.read(AsylumCaseDefinition.CASE_BUNDLES);
+
+        final List<Bundle> caseBundles = maybeCaseBundles.isPresent() ? maybeCaseBundles.get()
+            .stream()
+            .map(IdValue::getValue)
+            .collect(Collectors.toList()) : Collections.emptyList();
+
+        return caseBundles.isEmpty() ? "" : caseBundles.get(0).getStitchStatus().orElse("");
```

Comment:
it is ok for now, we need to think how to differentiate CASE_BUNDLES statuses when we move appeal skeleton bundle to async flow

Reply:
Agree Marcin. That would also depend on the UX design whether we plan to put alll bundles together or in separate collections. We can handle it later I think.